.github/workflows/scripts/get-codeowners.sh

```diff
@@ -9,15 +9,29 @@
 
 set -euo pipefail
 
+get_component_type() {
+  echo "${COMPONENT}" | cut -f 1 -d '/'
+}
+
 get_codeowners() {
-  echo "$((grep -m 1 "${1}" .github/CODEOWNERS || true) | sed 's/   */ /g' | cut -f3- -d ' ')"
+  echo "$((grep -m 1 "^${1}" .github/CODEOWNERS || true) | sed 's/   */ /g' | cut -f3- -d ' ')"
```

Comment:
```suggestion
  echo "$((grep -m 1 "^${1}/ " .github/CODEOWNERS || true) | sed 's/   */ /g' | cut -f3- -d ' ')"
```
Could we solve this issue by making this regex more strict? I think if we add this (and remove the extra `/` on line 46) it might achieve the same effect.

Reply:
Updated as requested, I also simplified the if/else at the end of the script, let me know if it's more clear or if I should revert.